In [1]:
!ls

biodiversity.ipynb  observations.csv  species_info.csv


In [2]:
import pandas as pd

In [33]:
obs_df = pd.read_csv('observations.csv')
spe_df = pd.read_csv('species_info.csv')

display(obs_df.head())
display(spe_df.tail())

MaxRowsObs = obs_df.scientific_name.count()
print('% Missing Data on Observations dataset:')
print((1 - obs_df.count() / MaxRowsObs) * 100)
print('')
MaxRowsSpe = spe_df.category.count()
print('% Missing Data on Species dataset:')
print((1 - spe_df.count() / MaxRowsSpe) * 100)

scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85

category         scientific_name  \
5819  Vascular Plant        Solanum parishii   
5820  Vascular Plant           Solanum xanti   
5821  Vascular Plant  Parthenocissus vitacea   
5822  Vascular Plant       Vitis californica   
5823  Vascular Plant     Tribulus terrestris   

                                           common_names conservation_status  
5819                                Parish's Nightshade                 NaN  
5820            Chaparral Nightshade, Purple Nightshade                 NaN  
5821        Thicket Creeper, Virginia Creeper, Woodbine                 NaN  
5822            California Grape, California Wild Grape                 NaN  
5823  Bullhead, Caltrop, Goathead, Mexican Sandbur, ...                 NaN

% Missing Data on Observations dataset:
scientific_name    0.0
park_name          0.0
observations       0.0
dtype: float64

% Missing Data on Species dataset:
category                0.000000
scientific_name         0.000000
common_names            0.000000
conservation_status    96.720467
dtype: float64
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5824 entries, 0 to 5823
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   category             5824 non-null   object
 1   scientific_name      5824 non-null   object
 2   common_names         5824 non-null   object
 3   conservation_status  191 non-null    object
dtypes: object(4)
memory usage: 182.1+ KB
None


In [39]:
print(spe_df.conservation_status.unique())
print(obs_df.info())
print(spe_df.info())
print(obs_df.dtypes)
print(spe_df.dtypes)

[nan 'Species of Concern' 'Endangered' 'Threatened' 'In Recovery']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23296 entries, 0 to 23295
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   scientific_name  23296 non-null  object
 1   park_name        23296 non-null  object
 2   observations     23296 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 546.1+ KB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5824 entries, 0 to 5823
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   category             5824 non-null   object
 1   scientific_name      5824 non-null   object
 2   common_names         5824 non-null   object
 3   conservation_status  191 non-null    object
dtypes: object(4)
memory usage: 182.1+ KB
None
scientific_name    object
park_name          object
observations        int64
dtype: object
cate

Primeras impresiones:

- En spe_df, la columna conservation_status presenta casi un 97% de falta de datos. Siendo tan poco representativa, habría que eliminarla. Las otras columnas si que tienen 100% de datos en todas sus filas, en ambos datasets. 
- En ambos datasets, hay que buscar valores duplicados.
- Hay que ver que clase de especies están incluidas. Es todo flora o también hay fauna? Se puede hacer una clasificación más general añadiendo una columna con valores como planta, mamifero, etc? O es un quebradero de cabeza inmenso buscar a que clase pertenecen todas las especies?
- En caso de querer realizar esta nueva columna, puede ser necesario cambiar el tipo de dato de algunas columnas a string. 
- Un gráfico de tipo mapa para los parques estaría brutal. Investigar cuando tenga más claros los objetivos.